internal/core/application/trade_service.go

```diff
@@ -734,21 +735,62 @@ func getPriceAndPreviewForMarket(
 	market *domain.Market,
 	tradeType int,
 	amount uint64,
-) (
-	price Price,
-	previewAmount uint64,
-	err error,
-) {
+	asset string,
+) (Price, uint64, error) {
+	balances := getBalanceByAsset(unspents)
+	baseAssetBalance := balances[market.BaseAsset]
+	quoteAssetBalance := balances[market.QuoteAsset]
+
+	if tradeType == TradeBuy {
+		if asset == market.BaseAsset && amount >= baseAssetBalance {
+			return Price{}, 0, errors.New("preovided amount is too big")
+		}
+	} else {
+		if asset == market.QuoteAsset && amount >= quoteAssetBalance {
+			return Price{}, 0, errors.New("preovided amount is too big")
+		}
+	}
+
 	if market.IsStrategyPluggable() {
-		previewAmount = calcPreviewAmount(market, tradeType, amount)
-		price = Price{
+		previewAmount := calcPreviewAmount(
+			market,
+			baseAssetBalance,
+			quoteAssetBalance,
+			tradeType,
+			amount,
+			asset,
+		)
+
+		price := Price{
 			BasePrice:  market.BaseAssetPrice(),
 			QuotePrice: market.QuoteAssetPrice(),
 		}
-		return
+		return price, previewAmount, nil
 	}
 
-	return previewFromFormula(unspents, market, tradeType, amount)
+	price, previewAmount, err := previewFromFormula(
+		market,
+		baseAssetBalance,
+		quoteAssetBalance,
+		tradeType,
+		amount,
+		asset,
+	)
+	if err != nil {
+		return Price{}, 0, err
+	}
+
+	if tradeType == TradeBuy {
+		if asset == market.QuoteAsset && previewAmount >= baseAssetBalance {
+			return Price{}, 0, errors.New("preovided amount is too big")
```

Comment:
```suggestion
			return Price{}, 0, errors.New("given amount is too big")
```

Reply:
I used the same message of the `marketmaking` pkg.